packages/dds/tree/src/shared-tree/schematizingTreeView.ts

```diff
@@ -301,33 +301,40 @@ export class SchematizingSimpleTreeView<
 			addConstraintsToTransaction(this.checkout, constraintsOnRevert, constraints);
 		};
 
-		this.checkout.transaction.start();
-
-		// Validate preconditions before running the transaction callback.
-		addConstraints(false /* constraintsOnRevert */, params?.preconditions);
-		const transactionCallbackStatus = transaction();
-		const rollback = transactionCallbackStatus?.rollback;
-		const value = (
-			transactionCallbackStatus as TransactionCallbackStatus<TSuccessValue, TFailureValue>
-		)?.value;
-
-		if (rollback === true) {
-			this.checkout.transaction.abort();
-			return value !== undefined
-				? { success: false, value: value as TFailureValue }
-				: { success: false };
-		}
+		const executeTransaction = ():
+			| TransactionResultExt<TSuccessValue, TFailureValue>
+			| TransactionResult => {
+			this.checkout.transaction.start();
 
-		// Validate preconditions on revert after running the transaction callback and was successful.
-		addConstraints(
-			true /* constraintsOnRevert */,
-			transactionCallbackStatus?.preconditionsOnRevert,
-		);
+			// Validate preconditions before running the transaction callback.
+			addConstraints(false /* constraintsOnRevert */, params?.preconditions);
+			const transactionCallbackStatus = transaction();
+			const rollback = transactionCallbackStatus?.rollback;
+			const value = (
+				transactionCallbackStatus as TransactionCallbackStatus<TSuccessValue, TFailureValue>
+			)?.value;
+
+			if (rollback === true) {
+				this.checkout.transaction.abort();
+				return value !== undefined
+					? { success: false, value: value as TFailureValue }
+					: { success: false };
+			}
+
+			// Validate preconditions on revert after running the transaction callback and was successful.
+			addConstraints(
+				true /* constraintsOnRevert */,
+				transactionCallbackStatus?.preconditionsOnRevert,
+			);
+
+			this.checkout.transaction.commit();
+
+			return value !== undefined
+				? { success: true, value: value as TSuccessValue }
+				: { success: true };
+		};
 
-		this.checkout.transaction.commit();
-		return value !== undefined
-			? { success: true, value: value as TSuccessValue }
-			: { success: true };
+		return this.checkout.runWithTransactionLabel(() => executeTransaction(), params?.label);
```

Comment:
I would slightly prefer that you just wrap directly with the function rather than binding the lamda to a variable. Since it's only used once. Up to you though.

```ts
runWithTransactionLabel(() => {
   // ...
});
```

rather than 

```ts
const executeTransaction = () => {
  // ...
};

runWithTransactionLabel() => executeTransaction);
```